### ISSUE TITLE
TagManager setup component show error message on settings page

### ISSUE THREAD

**Issue by bjackson27** [google/site-kit-wp#181]
Similar to what we have for GA in this situation (another admin set up the account), let's say
"You currently don't have access to this Google Tag Manager account. You can either request access from your team, or remove this Google Tag Manager snippet and connect to a different account."

Steps to reproduce:
1. Setup Site Kit as usual for the first administrator.
2. Setup Analytics and TagManager
3. Logout and login as second Administrator
4. Sign in to Google with an account that doesn't have access to the TagManager account
5. Go to settings page, and the error message show up.

![Screen Shot 2019-05-17 at 19 11 54](https://user-images.githubusercontent.com/28811945/61000121-53b38400-a322-11e9-8e19-17e11ee8d603.png)
---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* For cases where there's an existing GTM account connected through the plugin by another admin user, but the current admin user doesn't have access to it, display relevant messaging to make it clear what the status is and what their next action can be:
"You currently don't have access to this Google Tag Manager account. You can either request access from your team, or remove this Google Tag Manager snippet and connect to a different account."
* Similarly for Google Analytics configured by another admin: "You currently don't have access to this Google Analytics account. You can either request access from your team, or remove this Google Analytics snippet and connect to a different account."

## Implementation Brief

Rename `error` and `message` in state to `errorCode` and `errorMsg` to be consistent with other components in https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/tagmanager/setup.js

`errorCode` is needed to determine the styling the message should have: a regular notice or error format. We currently display this type of error as a notice in Analytics; we should have the same formatting in TagManager to be consistent.  Therefor, implement `renderErrorOrNotice()` used in Analytics setup to render as a regular message and not formatted as an error message (red text).

In [`requestTagManagerAccounts()`](https://github.com/google/site-kit-wp/blob/5ba7ab9ba7b8587d876fb2935166c53eed853b88/assets/js/modules/tagmanager/setup.js#L121), if `selectedAccount` is set, filter `responseData.accounts` looking for matching account id. 

If there is no matching ID, set state `errorCode` with the specific error code 'insufficientPermissions' and `errorMsg` with message provided in the issue description.

## Changelog entry

*

**Comment by lucymtc** [comment]
@felixarntz I have updated the Implementation brief, I believe we need to keep both `errorCode` and `errorMsg` so we can render notice or error like we do in Analytics setup component https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/analytics/setup.js#L874 I think we should keep this one as a notice to be consistent with the same scenario on analytics.

**Comment by felixarntz** [comment]
@lucymtc Makes sense, thanks!

**Comment by swissspidy** [comment]
PR approved, moved to QA column 👍

**Comment by adamsilverstein** [comment]
QAing this fix in the branch `fix/181-tagmanager-error-message` from https://github.com/google/site-kit-wp/pull/321... 

The new notice for the second admin looks good:

![image](https://user-images.githubusercontent.com/2676022/62064026-4256ec80-b1e9-11e9-912e-f19950b6c458.png)

As a user, I'm a little confused by this part of the notice:
>  or remove this Google Tag Manager snippet and connect to a different account.

How do I do that? Can we make that a bit clearer to the user?

One additional state we may want to address in a separate issue: in one test I logged in as the second admin and chose a Google account that didn't have a Tag Manager account (at all) and received an error message. Do we want to update the messaging here?

![image](https://user-images.githubusercontent.com/2676022/62063906-0885e600-b1e9-11e9-9469-e9269bd09683.png)

The API response code was `google_tagmanager_account_empty`

![image](https://user-images.githubusercontent.com/2676022/62064322-d2953180-b1e9-11e9-8474-cfeb42130d79.png)

**Comment by felixarntz** [comment]
> How do I do that? Can we make that a bit clearer to the user?

You'd need to edit the Tag Manager settings. Given the context, I'm not sure how to clarify this better. The other consideration is that we maybe shouldn't even make it too obvious: If the first admin set up a Tag Manager account, the secondary admin shouldn't just delete it because they don't have access. We should continue to emphasize the "you can either request access from your team".

> One additional state we may want to address in a separate issue: in one test I logged in as the second admin and chose a Google account that didn't have a Tag Manager account (at all) and received an error message. Do we want to update the messaging here?

Makes sense. I believe it should display the same message as above: If you don't have access to the currently set account, that's what matters, regardless of whether you have an account yourself or not.

**Comment by felixarntz** [comment]
@adamsilverstein I opened #346 to address the issue you noticed separately, as it's a separate (though probably small) bug. If everything else here is fine, feel free to approve this QA.

**Comment by adamsilverstein** [comment]
Thanks for the comments and review @felixarntz - I will retest this once the PR is merged to develop, then approve this QA.

**Comment by marrrmarrr** [comment]
LGTM
